halyard-web/src/main/java/com/netflix/spinnaker/halyard/controllers/v1/PersistentStorageController.java

```diff
@@ -57,19 +58,19 @@
       builder.setValidateResponse(() -> persistentStorageService.validatePersistentStorage(deploymentName));
     }
 
-    return DaemonTaskHandler.submitTask(builder::build, "Get storage settings");
+    return DaemonTaskHandler.submitTask(builder::build, "Get persistent storage settings");
   }
 
-  @RequestMapping(value = "/", method = RequestMethod.PUT)
-  DaemonTask<Halconfig, Void> setPersistentStorage(@PathVariable String deploymentName,
-      @RequestParam(required = false, defaultValue = DefaultControllerValues.validate) boolean validate,
-      @RequestParam(required = false, defaultValue = DefaultControllerValues.severity) Severity severity,
-      @RequestBody Object rawPersistentStorage) {
-    PersistentStorage persistentStorage = objectMapper.convertValue(rawPersistentStorage, PersistentStorage.class);
+  @RequestMapping(value = "/currentPersistentStore", method = RequestMethod.PUT)
+  DaemonTask<Halconfig, Void> setCurrentPersistentStore(@PathVariable String deploymentName,
+                                                        @RequestParam(required = false, defaultValue = DefaultControllerValues.validate) boolean validate,
+                                                        @RequestParam(required = false, defaultValue = DefaultControllerValues.severity) Severity severity,
+                                                        @RequestBody Object rawPersistentStoreName) {
+    String persistentStoreName = objectMapper.convertValue(rawPersistentStoreName, String.class);
```

Comment:
This endpoint breaks convention, usually you would pick a storage type by editing the top-level persistent storage object containing all storage types. This would be surfaced in halyard has `hal config storage edit --type azs`.

Take a look at the metric store controller to see how it's handled there.